src/liboslexec/llvm_util.cpp

```diff
@@ -1767,6 +1860,8 @@ LLVM_Util::setup_optimization_passes(int optlevel, bool target_host)
 
     m_llvm_module_passes           = new llvm::legacy::PassManager;
     llvm::legacy::PassManager& mpm = (*m_llvm_module_passes);
+    // TODO: Add based on optlevel
+    mpm.add(new CheckLayerRemovalPass());
```

Comment:
Since the new pass is being run on the entire module, I think that means that it's running on all of the shadeops functions. Would it be possible to detect that you are processing a library function (by function name, or perhaps a function attribute), and early-exit from `CheckLayerRemovalPass::runOnFunction()` in that case?

That might not buy much since module pruning should remove unused library functions, but maybe it will help in some cases.

Reply:
That seems like a responsible safe-guard to put in, I'll add that.